### PR TITLE
Better handling of axios errors

### LINKED ISSUE #1: The authenticate() function may fail with no console error message for certain error types
## Context
Found that for specific type of firewall error in my organization, the `authenticate()` would fail without showing the cause of the error.
The suppressed error for me was ` Error: unable to get local issuer certificate`.
I was able to expose the root cause by modifying the existing `console.error()` line. See below.

## Expected Behavior
The error message should be output the same as other related error messages so such as:
```
 Authentication error while importing Xray results, reason :
 { error: 'Authentication failed. Invalid client credentials!' }
```

## Actual Behavior
The console displays authentication progress but then exits and shows nothing and the Xray import did not take place
```
...
 ✔ Invalid Email Address | {"emailAddress":".boldy@boldcommece.com"} in 0ms

  OK  | 14 passed   // 10s
```

## Troubleshooting
I was able to modify [this line](https://github.com/saadichouaib/codeceptjs-xray-cloud-helper/blob/develop/api/xray_api.js#L38) to just print the raw error.  similar to something like:
```console.error(`Authentication error while importing Xray results, reason :\n ${err}`);```

And was able to get the following output:
```
Authentication error while importing Xray results, reason :
 Error: unable to get local issuer certificate
```

### PR DESCRIPTION
**Pull request by aminekun90**
Here is a better handling of the auth error I couldn't tests this, can you test it if it work thats perfect

### CODE REVIEW

**Comment by saadichouaib** [comment]
LGTM 🥇 

It would be perfect if we could do the same with `execute_import()` and `import_cucumber_feature()`.

We could put your solution in a separate function that gets error as an argument and returns error_message.

We could have something like :

`const api_error = require("../helpers/api_error");`
```
catch (error) {
     load_auth.stop();
     
     const error_message = api_error.handle_error(error);
     
     console.error(`Authentication error while importing Xray results, reason :\n`, error_message);
     process.exit(1);
}
```

**Comment by aminekun90** [comment]
there you go hope it's better now @saadichouaib

**Review by saadichouaib — APPROVED** [review]
Perfecto 💯

**Comment by jmychasiw** [comment]
Retesting under the same condition produces a better error message but still does not show root cause:
```
Authentication error while importing Xray results, reason :
 No response from the server : [object Object]
```

However modifying the console output to exclude the `request` object _(or wrap in JSON.stringify)_ produces:
```
            } else if (error.request) {
                // Request made but no response is received from the server.
                errorMessage = "No response from the server : " + error;
```

```
Authentication error while importing Xray results, reason :
 No response from the server : Error: unable to get local issuer certificate
```

**Comment by jmychasiw** [comment]
out of interest [this network security tool](https://www.zscaler.com/) is what is producing the error for me.  This is not a blocker as I can temporarily bypass and I am in the process of getting access approved.

**Comment by aminekun90** [comment]
Hey @jmychasiw thanks for the feedbacks, You should be able to get a better display of the error as the error handler returns an object containing {message,error} instead of a string 
FIY : @saadichouaib

File: api/api_errors.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module.exports = {
+    /**
+     * 
+     * @param {Error | *} error 
+     * @returns 
+     */
+    handle_axios_error(error) {
+        let errorMessage = {};
+        console.debug("Iitial error : ",error);
```
Comment (saadichouaib) [inline api/api_errors.js]: You used this line only for test ? do you think we can remove it since it prints the whole object?

**Comment by jmychasiw** [comment]
The new approach proved much more informative.
```
  message: 'No response from the server : ',
  error: AxiosError: unable to get local issuer certificate
      at Function.AxiosError.from (/Users/.../Workspace/xray-plugin-poc/node_modules/axios/lib/core/AxiosError.js:89:14)
      at RedirectableRequest.handleRequestError (/Users/../Workspace/xray-plugin-poc/node_modules/axios/lib/adapters/http.js:577:25)
      at RedirectableRequest.emit (node:events:539:35)
      at RedirectableRequest.emit (node:domain:475:12)
      at ClientRequest.eventHandlers.<computed> (/Users/.../Workspace/xray-plugin-poc/node_modules/follow-redirects/index.js:14:24)
      at ClientRequest.emit (node:events:527:28)
      at ClientRequest.emit (node:domain:475:12)
      at TLSSocket.socketErrorListener (node:_http_client:454:9)
      at TLSSocket.emit (node:events:527:28)
      at TLSSocket.emit (node:domain:475:12) {
    code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY',
    config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 120000,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [AxiosHeaders],
      method: 'post',
      url: 'https://xray.cloud.getxray.app/api/v2/authenticate',
      data: '{"client_id":"..","client_secret":".."}'
    },
    request: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _options: [Object],
      _ended: false,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 131,
      _requestBodyBuffers: [Array],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [ClientRequest],
      _currentUrl: 'https://xray.cloud.getxray.app/api/v2/authenticate',
      _timeout: null,
      [Symbol(kCapture)]: false
    },
    cause: Error: unable to get local issuer certificate
        at TLSSocket.onConnectSecure (node:_tls_wrap:1532:34)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at TLSSocket._finishInit (node:_tls_wrap:946:8)
        at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:727:12) {
      code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY'
    }
  }
}
```

**Comment by jmychasiw** [comment]
I noticed one thing that may be concern to some is that the new output displays the raw `client_id` and `client_secret` in the console. This may be an issue in a running pipeline. I was wondering if there was an option to mask those values?
```
config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 120000,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [AxiosHeaders],
      method: 'post',
      url: 'https://xray.cloud.getxray.app/api/v2/authenticate',
      data: '{"client_id":"CLIENT_ID","client_secret":"CLIENT_SECRET"}'
```

**Comment by aminekun90** [comment]
> I noticed one thing that may be concern to some is that the new output displays the raw `client_id` and `client_secret` in the console. This may be an issue in a running pipeline. I was wondering if there was an option to mask those values?
> 
> ```
> config: {
>       transitional: [Object],
>       adapter: [Array],
>       transformRequest: [Array],
>       transformResponse: [Array],
>       timeout: 120000,
>       xsrfCookieName: 'XSRF-TOKEN',
>       xsrfHeaderName: 'X-XSRF-TOKEN',
>       maxContentLength: -1,
>       maxBodyLength: -1,
>       env: [Object],
>       validateStatus: [Function: validateStatus],
>       headers: [AxiosHeaders],
>       method: 'post',
>       url: 'https://xray.cloud.getxray.app/api/v2/authenticate',
>       data: '{"client_id":"CLIENT_ID","client_secret":"CLIENT_SECRET"}'
> ```

I've added a new commit to sanitise the data and the message from sensitive information like client_id or client_secret in message or config.data inside axiosError object

**Comment by jmychasiw** [comment]
Retested in my environment.  Looks good!
```
config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 120000,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [AxiosHeaders],
      method: 'post',
      url: 'https://xray.cloud.getxray.app/api/v2/authenticate',
      data: '{"client_id":"*****","client_secret":"*****"}'
    },
```

**Review by saadichouaib — APPROVED** [review]

File: api/api_errors.js (hunk 1 above)
Comment (aminekun90) [reply api/api_errors.js]: you are right I've updated that

**Pull request merged**